README.org
Collision integrals

Package for calculating intermolecular collision integrals used in
transport coefficient calculations of low pressure gases.

Package offers
- Collision integrals for hard-sphere collisions
- Reduced collision integrals for the 6-12 Lennard-Jones potential.
  Multiple methods are offered:
  - Analytic expressions for Omega-11* and Omega22*
  - Interpolation of tables by Ivchenko et al
- Mechanism for choosing the default calculation method or to use a
  specific calculation method.
- Certain combinations of the reduced \Omega integrals, A^*, B^*, C^*, E^*, and
  F^*



ChangeLog
2013-04-07  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* collision-integrals-package-def.lisp (:collision-integrals):
	Exporting *omega*-table-1* and *omega*-table-2* instead of
	*omega*-table*

	* collision-parameters.lisp
	(make-lennard-jones-6/12-collision-parameters): Relaxed test
	criterion to 1e-6

2012-07-11  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* omega-r-setup.lisp: Added calls to initialize Omega* for table
	1B of Ivchenko et al
	(setf) : Modified setf to initialize defaults to Omega* from table
	1b of Ivchenko et al.

	* lj-collision-integral-table-interpolations.lisp (+t*-min+)
	(+t*-max+): Changed to double float
	(omega*-table): Removed column-alist initialization from class
	specifiation to the instantiation
	(make-omega*-table): Changed calling interface, added function
	documentation, and made the object documentation more descriptive
	(*omega*-table-1*) (*omega*-table-2*): Changed from variable into
	parameters.  Improved documentation
	(omega*%-11, omega*%-12, omega*%-13, omega*%-14, omega*%-15)
	(omega*%-22): Changed specializer from obsolete *omega*-table* to
	*omega*-table-1*
	(omega*%-23, omega*%-24, omega*%-25, omega*%-26, omega*%-33)
	(omega*%-44): Added new functions that use *omega*-table-2*

	* collision-integrals.asd (:collision-integrals-user): Added
	dependency on mv-gnuplot and mv-grid-utils (it will be loaded by
	collision-integrals, but I add it here as documentation)
	(:collision-integrals-user): Added omega-plots component

	* collision-integrals-user-package-def.lisp
	(:collision-integrals-user): Added new use packages mv-gnuplot and
	mv-grid

	* collision-integrals-package-def.lisp (:collision-integrals):
	Added Omega integrals from table 2b to list of exported symbols

	* omega-plots.lisp (plot-interpolations): Added new file that
	verifies reading and interpolation of Omega* tables by plotting
	tabulated and interpolated values

	* lj-collision-integral-table-interpolations.lisp: Added
	definitions of omega* functions stored in table F1b.  This
	necessitated changes documented next.

2012-07-06  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* lj-collision-integral-table-interpolations.lisp
	(make-omega*-table): Changed interface.  It now has one required
	argument, the file
	(*omega*-table-1*, *omega*-table-2*): Instead of *omega*-table*, I
	now have two variables, each storing one table

2012-05-16  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* lj-collision-integral-table-interpolations.lisp
	(with-verbose-table-interpolation): 
	(omega*%-11, omega*%-12, omega*%-13, omega*%-14, omega*%-15)
	(omega*%-22): Added error handling when t*
	is outside of Ivchenko's table domain
	(with-verbose-table-interpolation): Envornment for handling out-of
	range T*

2012-05-15  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* collision-integrals.asd (:collision-integrals): module loading
	dependency on module "package"

2012-05-13  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* collision-integrals-user.lisp (demo): Fixed spelling error

	* omega-star-combinations.lisp: Added omega-* combination
	functions A*, B*, C*, E* and F*

2012-04-25  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* collision-parameters.lisp (hard-sphere-collision-parameters):
	Bug in class inheritance.  I had incorrectly specified
	lennar-jones-6/12 instead of hard-sphere


2011-10-16  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* collision-parameters.lisp (make-collision-parameters): Fixed bug
	in calculating sigma for same-species collisions.  It was twice as
	large as it should have been

	* omega-hs.lisp (omega-11-hs%): Modified numerical constants to
	agree with (Ferziger & Kaper, 9.2-4)

2011-09-15  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* omega-hs.lisp (omega-11-hs%): Changed argument units from SI to
	amu angstrom

2011-09-13  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* collision-parameters.lisp (collision-parameters): Changed object
	layout of collision parameters.  Work still in progress

	* omega-gf+defaults.lisp (defomega-ij): Changed omega-ij-hs calls
	to the ew object based omega-ij-hs calling convention

	* omega-hs.lisp Complete rewrite of the file.  The omega-xy-hs are
	now generic functions with one method that specializes on the
	objects of the hard-sphere-potential class.  With two helper
	fuctions, a macro is used to ease the definition of the generic
	functions and their methods.

2011-09-11  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* collision-parameters.lisp
	(make-lennard-jones-6/12-collision-parameters): Changed code to
	use the potential objects from `molecular-potentials' to retreive
	potential parameters

	* collision-integrals.asd (:collision-integrals): Added file
	`collision-parameters' to the system

2011-09-10  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* omega-gf+defaults.lisp (:omega-xx): Changing collision integrals
	to depend not on molecular potentials, but on collision
	parameters, which can be different from the molecular potentials.
	This will require a revamp of part of the system, and also a
	change to molecular-potentials system.

	* doc/collision-integrals-cl-package.org: Fixing argument order:
	mass, temperature, rest in documentation and code.

	* omega-hs.lisp: Fixed multiple bugs introduced by the change in
	parameter oredring from temperature mass sigma to mass temperature
	sigma.  The bug was detected by unit test failures

2011-09-06  Mirko Vukovic  <mirko.vukovic@us.tel.com>

	* lj-collision-integral-fits.lisp (omega*%-11, omega*%-22): Fixed
	typos in arguments due to change in naming convenction

	* omega-gf+defaults.lisp (defomega-ij): Verified definition of
	`omega-ij' as `omega*-ij * omega-ij-hs'

	* lj-default-methods.lisp (default-omega*-lj-calc-method)
	(set-default-omega*-lj-calc-method): Changed argument name from

omega-r-setup.lisp
;; Mirko Vukovic
;; Time-stamp: <2012-07-11 18:09:09 omega-r-setup.lisp>
;; 
;; Copyright 2011 Mirko Vukovic
;; Distributed under the terms of the GNU General Public License
;; 
;; This program is free software: you can redistribute it and/or modify
;; it under the terms of the GNU General Public License as published by
;; the Free Software Foundation, either version 3 of the License, or
;; (at your option) any later version.
;; 
;; This program is distributed in the hope that it will be useful,
;; but WITHOUT ANY WARRANTY; without even the implied warranty of
;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
;; GNU General Public License for more details.
;; 
;; You should have received a copy of the GNU General Public License
;; along with this program.  If not, see <http://www.gnu.org/licenses/>.

(in-package :omega-xx)

;; setup the generic functions so that we can add methods in the
;; methods module.  Also set-up the default calling functions


(defmacro defomega*-ij (i j)
  "Expands into forms that define:
- special variable that holds the default calculation method for omega*-ij
- generic function omega-ij*%
- function omega-ij*

For details, of expansion, see unit test defomega*-ij"
  (let ((si (prin1-to-string i))
	(sj (prin1-to-string j)))
    (let ((gf-name (alexandria:symbolicate "OMEGA*%-" si sj))
	  (default-name (alexandria:symbolicate "*OMEGA*-" si sj "-DEFAULT*"))
	  (f-name (alexandria:symbolicate "OMEGA*-" si sj)))
      `(progn
	 (defvar ,default-name nil
	   ,(format nil "Default method for calculating omega*-~a~a" i j))
	 (defgeneric ,gf-name (model T*)
	   (:documentation
	    ,(format nil "Evaluate the omega-~a~a `model' at reduced-temperature T*
If model is specified as `t', use default model" i j))
	   (:method ((model (eql t)) reduced-temperature)
	     (funcall #',gf-name ,default-name reduced-temperature)))
	 (defun ,f-name (T*)
	   ,(format nil "Evaluate omega-~a~a at T* using the default method" i j)
	   (,gf-name t T*))))))

(define-test defomega*-ij
  (assert-expands
   '(progn
     (defvar *omega*-11-default* nil
       "Default method for calculating omega*-11")
     (defgeneric omega*%-11 (model T*)
      (:documentation "Evaluate the omega-11 `model' at reduced-temperature T*
If model is specified as `t', use default model")
      (:method ((model (eql t)) reduced-temperature)
	(funcall #'omega*%-11 *omega*-11-default* reduced-temperature)))
     (defun omega*-11 (T*)
       "Evaluate omega-11 at T* using the default method"
       (omega*%-11 t T*)))
   (defomega*-ij 1 1 )))

(defomega*-ij 1 1)
(defomega*-ij 1 2)
(defomega*-ij 1 3)
(defomega*-ij 1 4)
(defomega*-ij 1 5)
(defomega*-ij 2 2)
(defomega*-ij 2 3)
(defomega*-ij 2 4)
(defomega*-ij 2 5)
(defomega*-ij 2 6)
(defomega*-ij 3 3)
(defomega*-ij 4 4)

(setf *omega*-11-default* *omega*-table-1*
      *omega*-12-default* *omega*-table-1*
      *omega*-13-default* *omega*-table-1*
      *omega*-14-default* *omega*-table-1*
      *omega*-15-default* *omega*-table-1*
      *omega*-22-default* *omega*-table-1*
      *omega*-23-default* *omega*-table-2*
      *omega*-24-default* *omega*-table-2*
      *omega*-25-default* *omega*-table-2*
      *omega*-26-default* *omega*-table-2*
      *omega*-33-default* *omega*-table-2*
      *omega*-44-default* *omega*-table-2*
      )